google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploymentConfigurator.java

```diff
@@ -115,6 +155,24 @@ public AppEngineDeploymentConfiguration createDefaultConfiguration(
     return new AppEngineDeploymentRunConfigurationEditor(
         project,
         source,
-        new CloudSdkAppEngineHelper(new File(server.getConfiguration().getCloudSdkHomePath())));
+        new CloudSdkAppEngineHelper(
+            project,
+            new File(server.getConfiguration().getCloudSdkHomePath()),
+            getEnvironment())
+    );
+  }
+
+  /**
+   * Lazily initialize the environment because this class is instantiated on IDE boot time
+   * and the project may not have yet been fully constructed.
+   */
+  public Environment getEnvironment() {
```

Comment:
Facets are at the module level though aren't they?

On Fri, Jun 3, 2016 at 1:10 PM, Etan Shaul <notifications@github.com> wrote:

> In
> google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploymentConfigurator.java
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/700#discussion_r65742521>
> :
>
> > @@ -115,6 +155,24 @@ public AppEngineDeploymentConfiguration createDefaultConfiguration(
> >      return new AppEngineDeploymentRunConfigurationEditor(
> >          project,
> >          source,
> > -        new CloudSdkAppEngineHelper(new File(server.getConfiguration().getCloudSdkHomePath())));
> > +        new CloudSdkAppEngineHelper(
> > +            project,
> > +            new File(server.getConfiguration().getCloudSdkHomePath()),
> > +            getEnvironment())
> > +    );
> > +  }
> > +
> > +  /**
> > +   * Lazily initialize the environment because this class is instantiated on IDE boot time
> > +   * and the project may not have yet been fully constructed.
> > +   */
> > +  public Environment getEnvironment() {
>
> Currently, it determined based on the project itself: if the project has
> the AE standard facet from the old plugin, then its standard. Otherwise,
> its flex. This came from the document that we went over earlier, but is
> obviously open for change.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/700/files/cf9d207bfecca398c61805abbbdf7ec498e8e5fc#r65742521>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHf5HbcBvHuGaESrKtIjqS5mNn5eDdVFks5qIF_ogaJpZM4Im4aR>
> .
>